libs/dal/src/repositories/layout/layout.repository.ts

```diff
@@ -0,0 +1,87 @@
+import { AuthProviderEnum } from '@novu/shared';
+import { FilterQuery } from 'mongoose';
+
+import { LayoutEntity } from './layout.entity';
+import { Layout } from './layout.schema';
+import { EnvironmentId, ExternalSubscriberId, OrganizationId, LayoutId, LayoutName } from './types';
+
+import { BaseRepository, Omit } from '../base-repository';
+
+class PartialIntegrationEntity extends Omit(LayoutEntity, ['_environmentId', '_organizationId']) {}
+
+type EnforceEnvironmentQuery = FilterQuery<PartialIntegrationEntity> &
+  ({ _environmentId: EnvironmentId } | { _organizationId: OrganizationId });
+
+export class LayoutRepository extends BaseRepository<EnforceEnvironmentQuery, LayoutEntity> {
+  constructor() {
+    super(Layout, LayoutEntity);
+  }
+
+  async createLayout(entity: Omit<LayoutEntity, '_id'>): Promise<LayoutEntity> {
+    const { channel, content, contentType, isDefault, name, variables, _creatorId, _environmentId, _organizationId } =
+      entity;
+
+    return await this.create({
+      _creatorId,
+      _environmentId,
+      _organizationId,
+      content,
+      contentType,
+      isDefault,
+      isDeleted: false,
+      name,
+      variables,
+    });
+  }
+
+  async deleteLayout(_id: LayoutId, _environmentId: EnvironmentId, _organizationId: OrganizationId): Promise<void> {
+    await this.update(
+      {
+        _id,
+        _environmentId,
+        _organizationId,
+      },
+      {
+        isDeleted: true,
+      }
+    );
+  }
```

Comment:
The plugin also allows here to to search for a record and then just do this.layout.delete